pkgs/development/tools/mcap/mcap-cli/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "mcap-cli";
+  version = "0.0.28";
+
+  src = fetchFromGitHub {
+    owner = "foxglove";
+    repo = "mcap";
+    rev = "fd3ecada218f41929c6d430acba3561e5ebfaf08";
+    sha256 = "sha256-pIZThgb0wRmhNi3YrYTC+Tr6a/4Q7CWuHw+icjb1p8g=";
+  } + "/go/cli/mcap";
```

Comment:
Using `sourceRoot` would be more idiomatic.

Reply:
When I use `sourceRoot = "source/go/cli/mcap"` I get the following error when `buildGoModule` runs `go vendor`: 
```
go: -mod may only be set to readonly when in workspace mode, but it i>
        Remove the -mod flag to use the default readonly value,
        or set GOWORK=off to disable workspace mode.
```

Reply:
```diff
diff --git a/pkgs/development/tools/mcap/mcap-cli/default.nix b/pkgs/development/tools/mcap/mcap-cli/default.nix
index b6ae32e563f..5cf9459130e 100644
--- a/pkgs/development/tools/mcap/mcap-cli/default.nix
+++ b/pkgs/development/tools/mcap/mcap-cli/default.nix
@@ -12,10 +12,14 @@ buildGoModule rec {
     repo = "mcap";
     rev = "fd3ecada218f41929c6d430acba3561e5ebfaf08";
     sha256 = "sha256-pIZThgb0wRmhNi3YrYTC+Tr6a/4Q7CWuHw+icjb1p8g=";
-  } + "/go/cli/mcap";
+  };
+
+  sourceRoot = "source/go/cli/mcap";
 
   vendorHash = "sha256-3TSwOgTIaBg5U8UdQ74LmsS8gS05Yw02JjfaIN9fKQQ=";
 
+  GOWORK = "off";
+
   # tests are currently failing because the test mcap files are in another directory in the repo
   doCheck = false;
 

```